skel/daemons/settings.go

```
@@ -0,0 +1,368 @@
+	sort.SliceStable(currencies, func(i, j int) bool {
+		return currencies[i].ParentID < currencies[j].ParentID
+	})
+
+	for _, currency := range currencies {
+		if currency.Type == "coin" {
+			if currency.ParentID == "" {
+				res[currency.ID] = append(res[currency.ID], currency.ID)
+			} else {
+				res[currency.ParentID] = append(res[currency.ParentID], currency.ID)
+			}
+		}
+	}
+
+	return res
+}
+
+// Method createDepositAndHotWallet creates deposit and hot wallet
+func createDepositAndHotWallet(peatioClient *peatio.Client, currencies []string, opendaxAddr string) {
+	// Create Deposit Wallet
+	depositWalletParams := peatio.CreateWalletParams{
+		BlockchainKey: "opendax-cloud",
+		Name:          fmt.Sprintf("%s Deposit Wallet", strings.ToUpper(currencies[0])),
+		Kind:          "deposit",
+		Gateway:       "opendax_cloud",
+		Address:       "address",
+		Currencies:    currencies,
+		Status:        "active",
+		Settings: peatio.Settings{
+			URI: fmt.Sprintf("%v/api/v2/opx/peatio", opendaxAddr),
+		},
+	}
+
+	_, depositApiError := peatioClient.CreateWallet(depositWalletParams)
+	if depositApiError != nil {
+		log.Printf("ERROR: createWallets: Can't create deposit wallet. Error: %v. Errors: %v", depositApiError.Error, depositApiError.Errors)
+		return
+	}
+
+	// Create Hot Wallet
+	hotWalletParams := peatio.CreateWalletParams{
+		BlockchainKey: "opendax-cloud",
+		Name:          fmt.Sprintf("%s Hot Wallet", strings.ToUpper(currencies[0])),
+		Kind:          "hot",
+		Gateway:       "opendax_cloud",
+		Address:       "address",
+		Currencies:    currencies,
+		Status:        "active",
+		Settings: peatio.Settings{
+			URI: fmt.Sprintf("%v/api/v2/opx/peatio", opendaxAddr),
+		},
+	}
+
+	_, hotApiError := peatioClient.CreateWallet(hotWalletParams)
+	if hotApiError != nil {
+		log.Printf("ERROR: createWallets: Can't create deposit wallet. Error: %v. Errors: %v", hotApiError.Error, hotApiError.Errors)
+		return
```

Same, we should return an error here

---

we cant return errors here, as it will stop creating other wallets in the loop